Fixes for unicode in transfer names

Hwesta opened the pull request:
- Upgrade slumber so it doesn't use deprecated simplejson library (Edit: This won't be needed with #550)
- Encode filesystem paths as bytestrings
- In JS, assume strings are unicode and can be utf8 encoded.  Note that this probably removes support for non-unicode transfer names and paths, but needs further testing

refs #9234

Edit:
- Add MCPServer bugfix for unicode (from database) vs str (from disk) paths (Edit: Merged as part of #545)

Hwesta committed:
Upgrade slumber, removes simplejson dependency

Hwesta committed:
JS: base64 always assumes needs escaping

Hwesta committed:
Filesystem AJAX: paths should always be bytestrings

sevein commented on src/dashboard/src/requirements/base.txt:
```diff
@@ -15,7 +15,7 @@ git+https://github.com/artefactual-labs/agentarchives.git#egg=agentarchives
 mysqlclient==1.3.7
 # Required by storage-service component
-slumber==0.6.0
+slumber
```
Can you pin the dependency to the specific version you were testing?

sevein commented:
Holly, what's the difference between this `base64.js` and the base64-helpers submodule?

Hwesta committed:
Unit: Handle unicode in file paths

When matching between the file path on disk and the currentlocation stored in
the database, ensure that both are bytestrings.  By default, paths coming
from the database are unicode.  This fixes errors in finding filenames with
unicode between when they're assigned a UUID and when filename sanitization
happens.

Hwesta commented:
Added some related unicode work. This fixes having filenames with unicode in the name (something I thought worked already?).  It was failing to match the path-on-disk with the path-in-the-db because of a unicode/str mismatch.  Encode the unicode as utf8 for the match.

This will probably not work with non-unicode filenames.

Hwesta commented:
We haven't run into this issue recently, even though this fix hasn't been merged. This may not be required.

jhsimpson commented:
I am closing this PR, we are not going to merge it now.  There is an issue that has been opened on the acceptance-tests repo to define user stories for unicode transfer names, so we know what needs to be supported.  https://github.com/artefactual-labs/archivematica-acceptance-tests/issues/11
